dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java

```diff
@@ -16,42 +16,121 @@
  */
 public enum DescriptorLanguage {
     // Add new descriptor language here
-    CWL("CWL", "Common Workflow Language"), WDL("WDL", "Workflow Description Language"), NEXTFLOW("NFL", "Nextflow");
+    CWL("CWL", "Common Workflow Language", FileType.DOCKSTORE_CWL, FileType.CWL_TEST_JSON),
+    WDL("WDL", "Workflow Description Language", FileType.DOCKSTORE_WDL, FileType.WDL_TEST_JSON),
+    // DOCKSTORE-2428 - demo how to add new workflow language
+    //SWL("SWL", "Silly Workflow Language", FileType.DOCKSTORE_SWL, FileType.SWL_TEST_JSON)
+    NEXTFLOW("NFL", "Nextflow", FileType.NEXTFLOW_CONFIG, FileType.NEXTFLOW_TEST_PARAMS);
 
     public static final String CWL_STRING = "cwl";
     public static final String WDL_STRING = "wdl";
     public static final String NFL_STRING = "nfl";
+    // DOCKSTORE-2428 - demo how to add new workflow language
+    // public static final String SWL_STRING = "swl";
     /**
      * this name is used in the workflow path
      */
-    private final String descriptorLanguageShort;
+    private final String shortName;
 
     /**
      * this name is what is displayed to users to name the descriptor language
      */
     private final String friendlyName;
 
-    DescriptorLanguage(final String descriptorLanguageShort, final String friendlyName) {
-        this.descriptorLanguageShort = descriptorLanguageShort;
+    /**
+     * This is the primary descriptor filetype stored for files of this language in the database
+     */
+    private final FileType fileType;
+
+    /**
+     * This is the type for the test parameter file for this language
+     */
+    private final FileType testParamType;
+
+    DescriptorLanguage(final String shortName, final String friendlyName, final FileType fileType, final FileType testParamType) {
+        this.shortName = shortName;
         this.friendlyName = friendlyName;
+        this.fileType = fileType;
+        this.testParamType = testParamType;
     }
 
     @Override
     public String toString() {
-        return descriptorLanguageShort;
+        return shortName;
     }
 
     public String getFriendlyName() {
         return friendlyName;
     }
 
+    public static DescriptorLanguage convertShortStringToEnum(String descriptor) {
+        String lowerDescriptor = descriptor.toLowerCase();
+        switch (lowerDescriptor) {
+        case CWL_STRING:
+            return CWL;
+        case WDL_STRING:
+            return WDL;
+        case NFL_STRING:
+            return NEXTFLOW;
+        // DOCKSTORE-2428 - demo how to add new workflow language
+        //
+        //case SWL_STRING:
+        //    return SWL;
+        default:
+            // fall-through and throw exception
+        }
+        throw new UnsupportedOperationException("language not supported yet");
+    }
+
+    public FileType getFileType() {
+        return fileType;
+    }
+
+    public static Optional<FileType> getFileType(String descriptorType) {
+        for (DescriptorLanguage lang : values()) {
```

Comment:
You could alternatively write this and line 114 method using a stream and `findFirst()`. Don't know that's better though, unless we're making a conscious effort to go that way.

Reply:
Worth doing for fun